### ISSUE TITLE
dayBoundaries not provide to set start time greater than end time

### ISSUE THREAD

**Issue by maha-oueghlani** [tomosterlund/qalendar#117]
Hello, 
firstly thank you for this helpful component.

The issue that is made is in configuration of `dayBoundaries` : I want to set times begin from `08:00` to `07:00`  of the **next day** or from any start time that can be `>` than end time , and it is not possible with this version of `dayBoundaries`

**Reproduction URL**

you can see [here](https://stackblitz.com/edit/vue-57m8hr?file=src/components/HelloWorld.vue) a reproduction of this issue

**Comment by tomosterlund** [comment]
Hi @maha-oueghlani thanks for submitting this feature request. I agree, this would also be a good feature.I'll look into the technical requirements for achieving this, and will get back to you on whether I can picture implementing it.

**Comment by tomosterlund** [comment]
Hi @maha-oueghlani, I started working on this feature now (still got a few problems to tackle). One thing where I'm not quite sure on how to proceed, is how to display the dates in the week timeline. Please check the video below: what would be the preferred way for you:

As it is now: MO 6, TU 7, WE 8...
The more correct, but potentially a bit verbose way: MO-TU 6-7, TU-WE 7-8...


https://user-images.githubusercontent.com/63011648/224484893-4af7515f-697c-4777-b236-980558c5e882.mov

**Comment by maha-oueghlani** [comment]
> Hi @maha-oueghlani, I started working on this feature now (still got a few problems to tackle). I got two questions:
> 
>     1. One thing where I'm not quite sure on how to proceed, is how to display the dates in the week timeline. Please check the video below. What would be the preferred way for you?
> 
> 
> As it is now: MO 6, TU 7, WE 8... The more precise, but potentially a bit verbose way: MO-TU 6-7, TU-WE 7-8...
> 
>     2. If I release this feature as a beta version via npm, could you picture testing this in your application? I have to make adjustments in some very critical places of the code, and would probably not feel very comfortable releasing this without having one or two implementers test it on their use case.

Hello @tomosterlund 
1. I think that display dates like you are doing : MO 6, TU 7, WE 8... is better, then user can understand with timeline that if exceed 00:00 is the next day.
2. Sure, I will test it on my app and I will back to you with a video demo.

**Comment by tomosterlund** [comment]
@maha-oueghlani this feature is now available as a beta release with `npm i qalendar@2.4.0-beta.2`

**Comment by maha-oueghlani** [comment]
Hi @tomosterlund, I tested this version and overall it works great (please check the video), there are some issues which I will explain after the video.

https://user-images.githubusercontent.com/118192690/227975257-eada8b38-f02d-415c-bf89-7427565238c1.mp4

<br>
In this exemple I have the configuration like this : 

```
const config = {
    week: {
      startsOn: "monday",
      nDays: 7,
      scrollToHour: 0,
    },
     locale: "fr-FR", 
    defaultMode: "week",
    dayBoundaries: {
      start: 6,
      end: 6, // if end = 5,  if I have an event that start at 05:30 and finish at 06:30 it can't be shown, so this solution works 
    },
}
```
1- first issue : when I click on an event, an error is thrown : `Uncaught ReferenceError: TouchEvent is not defined` (check the video).
2- the second issue is : `scrollToHour`  is not working properly, here I set it to `0` but it scroll to `06:00` (strat hour), when I try to change it to another values it scrolls to another hour that makes no sense with the value set.

**Comment by maha-oueghlani** [comment]
> @maha-oueghlani this feature is now available as a beta release with `npm i qalendar@2.4.0-beta.2`
> 
> You might want to check out https://github.com/tomosterlund/qalendar/blob/feat/fluent_day_boundaries/adr/001-flexible-day-boundaries.md#consequences to see if this implementation fits what you would expect from the feature. If needed, we might build a config-option for overriding the behavior of consequence 3 & 4. But my assumption was that this was mainly about how events would be displayed in the day & week modes.

Hello @tomosterlund ,

for point 4, I think if you can add a configuration to allow the user to choose between displaying events on the week header or on the timeline.
Here (check the image) when the user will see the event, he will think there is a 2 day event, but it is only for one or two hours. If it can be displayed on timeline from `23:30` to `01:00` it is more proper and clearer.

Have a nice day :)

**Comment by tomosterlund** [comment]
@maha-oueghlani thanks a lot for testing and leaving feedback! This was very helpful. I'll look over the errors above.

For your last reply, could it be that you wanted to upload an image with it? Where it says "check the image" I don't see any.

**Comment by maha-oueghlani** [comment]
@tomosterlund  oh yes I forgot it sorry, 

I edited the comment.

**Comment by tomosterlund** [comment]
Ah thanks! This is also a bug. If `dayBoundaries: { start: 6, end: 4 }`, then an event like 23:30 - 01:00 should definitely be shown in one only day instead as a two day event. I'll fix this

**Comment by tomosterlund** [comment]
@maha-oueghlani fixes for the above 3 bugs + the feature requested from this issue is now available with `npm i qalendar@3.0.0-beta.1`

I figured a config-option to display an event in the week header, which could be displayed within a single day, is not needed. The reasonable behavior would be what you suggested.

**Comment by maha-oueghlani** [comment]
Thanks @tomosterlund  for your great work.
while waiting for the complete version 3, I will work with this awesome component in a dev environment.

**Comment by tomosterlund** [comment]
This feature is now available with v3.0.0, and documented at: https://tomosterlund.github.io/qalendar/guide.html#day-boundaries

Thanks again for opening this issue, and helping out with the testing!
